### DISCUSSION TITLE
Password Login Fails

### DISCUSSION THREAD

**Discussion by luiscvega** [singpass/demo-app#11] [Q&A] [answered]
Hello, I joined the early onboarding today and started working on the integration.

Password login is failing, even with the personas provided in [Demo App README](github.com/singpass/demo-app) and the [Myinfo Test Personas](https://docs.developer.singpass.gov.sg/singpass-developer-docs/technical-specifications/myinfo-test-personas).

Here is the error page:

![image](https://github.com/user-attachments/assets/338e89b8-6fe2-4bb1-8023-dffe73d7d5b7)

Here is my constructed `/auth` URL:

```
https://stg-id.singpass.gov.sg/auth?client_id=sOrxSWE9iShH01P3zTUPhwBLxrwi74qK&code_challenge=IDy1Jhf6PfVwwp_C_Jqed2FgHMBi4sYXS4Eeg3itQOA&code_challenge_method=S256&nonce=3d92ccf8-8dba-4271-be25-92012a36b869&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Fredirect&response_type=code&scope=openid+uinfin+name&state=19ad6148-1560-4a06-af56-070e8bd36a1c
```

Can anyone explain what's going on?

**Comment by luiscvega** [comment]
I was able to enter the Singpass ID and password earlier.

Now it's not even showing the login page anymore. It just says `invalid_client` now. 

![image](https://github.com/user-attachments/assets/9603f3df-2c03-4954-8305-3d743dd599dd).

Here is my `/auth` URL:

```
https://stg-id.singpass.gov.sg/auth?client_id=sOrxSWE9iShH01P3zTUPhwBLxrwi74qK&code_challenge=g_khdX_PuHyQgh9i0_9jSvPgZpdmE7q0uX0GLMee7g0&code_challenge_method=S256&nonce=5f143b49-f1e7-4eb6-a16c-7e3e564628ca&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Fredirect&response_type=code&scope=openid+uinfin+name&state=1dcaa0d1-1adc-4387-935b-8dff276273fb
```

Is the early onboarding not yet ready?

**Reply by poopypoops** [reply]
Hiii~ The 'invalid client' error can happen if Singpass OIDC can't retrieve your JWKS keys. Can you check if your JWKS JSON endpoint is up and working? You can use this tool to check the JWKS URL (if you're using one): https://poopypoops.github.io/Singpass-Login/. (created it for fun)

you can also double-check your SSL/TLS certificates to see if they're configured correctly and not expired here: https://www.ssllabs.com/ssltest/analyze.html. it could be because your ssl certs aren't configured correctly

**Reply by luiscvega** [reply]
Hi @poopypoops, thank you for the reply.

I see, thanks. I decided to use the JWKS Object for now and provided it with the JSON. Now the login page is now showing again.

However, I'm now running into the issue of the password login not working. I tried using the following Singpass ID/Password: `F1612351W / MyInfo2o15` but I'm getting the following error:

![Screenshot 2024-10-15 at 07-44-10 Singpass login](https://github.com/user-attachments/assets/e6bdb529-c4b8-49f4-8872-a92b755c9631)

**Reply by poopypoops** [reply]
Hi i think they updated the account!

**Comment by luiscvega** [comment, marked as answer]
This is fixed. Singpass has updated the passwords from `MyInfo2o15` to `Userinfo@2024`. Here is the [list of test personas](https://2816701917-files.gitbook.io/~/files/v0/b/gitbook-x-prod.appspot.com/o/spaces%2FW3T7d7fy7OGYkZf4zVKU%2Fuploads%2FMdbX65DlXWvq21Wsh8G7%2FUserinfo%20Test%20Accounts.csv?alt=media&token=63d551fb-daca-4faf-a192-1de82c9cabe6).
